src/java.base/share/classes/sun/reflect/generics/reflectiveObjects/TypeVariableImpl.java

```diff
@@ -125,18 +127,24 @@ public Type[] getBounds() {
     }
 
     /**
-     * Returns the {@code GenericDeclaration} object representing the
+     * Returns a {@code GenericDeclaration} object representing the
      * generic declaration that declared this type variable.
      *
-     * @return the generic declaration that declared this type variable.
+     * @return a generic declaration that declared this type variable.
      *
      * @since 1.5
      */
+    @SuppressWarnings("unchecked")
     public D getGenericDeclaration() {
         assert genericDeclaration instanceof Class<?> ||
                 genericDeclaration instanceof Method ||
                 genericDeclaration instanceof Constructor : "Unexpected kind of GenericDeclaration";
-        return genericDeclaration;
+        // If the `genericDeclaration` instance is mutable, we need to make a copy.
```

Comment:
Additionally, the old version would leak the internal root `Method` and `Constructor` objects.